op-e2e/inbox/inbox_test.go

```diff
@@ -0,0 +1,154 @@
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	ctx, _          = context.WithTimeout(context.Background(), 20*time.Second)
+	cost            = big.NewInt(1500000000000000)
+	depositVal      = new(big.Int).Mul(cost, big.NewInt(1000))
+	mockStorageAddr = common.Address{}
+)
+
+func TestBatchInboxFunctionSuccess(t *testing.T) {
+	op_e2e.InitParallel(t)
+
+	sys, l1Client := startSystemWithBatchInboxContract(t)
+	t.Cleanup(sys.Close)
+
+	// Wait for batch submitted and check event
+	requireEventualBatcherTx(t, &sys.Cfg, l1Client, 12*time.Second)
+}
+
+func startSystemWithBatchInboxContract(t *testing.T) (*e2esys.System, *ethclient.Client) {
+	cfg := e2esys.DefaultSystemConfig(t)
+	cfg.DataAvailabilityType = batcherFlags.BlobsType
+	cfg.BatcherTargetNumFrames = eth.MaxBlobsPerBlobTx
+	cfg.DeployConfig.UseInboxContract = true
+	c, ok := cfg.Nodes["sequencer"]
+	require.True(t, ok, "sequencer is required")
+	c.SafeDBPath = t.TempDir()
+	c.Driver.SequencerEnabled = true
+
+	sys, err := cfg.Start(t, e2esys.StartOption{
+		Key: "afterL1Start",
+		Action: func(cfg *e2esys.SystemConfig, s *e2esys.System) {
+			l1Client := s.NodeClient(e2esys.RoleL1)
+			// Deploy mock storage contract
+			mockStorageAddr = deployContract(t, cfg, l1Client, bindings.MockEthStorageMetaData, cost)
+			// Deploy BatchInbox.sol contract
+			batchInboxAddr := deployContract(t, cfg, l1Client, bindings.BatchInboxMetaData, mockStorageAddr)
+			t.Logf("mock storage %s, batchInbox %s, value %d", mockStorageAddr.Hex(), batchInboxAddr.Hex(), depositVal)
+			// Set BatchInboxAddress
+			cfg.DeployConfig.BatchInboxAddress = batchInboxAddr
+			// Deposit token
+			transferNativeTokenToBatchInboxAddress(t, cfg, l1Client, depositVal)
+		},
+	})
+	require.Nil(t, err, "Error starting up system")
+	return sys, sys.NodeClient(e2esys.RoleL1)
+}
+
+func requireEventualBatcherTx(t *testing.T, cfg *e2esys.SystemConfig, l1Client *ethclient.Client, timeout time.Duration) {
+	require.Eventually(t, func() bool {
+		for {
+			b, err := l1Client.BlockByNumber(ctx, nil)
+			require.NoError(t, err)
+			for _, tx := range b.Transactions() {
+				if tx.To().Cmp(cfg.DeployConfig.BatchInboxAddress) != 0 {
+					continue
+				}
+				receipt, err := l1Client.TransactionReceipt(ctx, tx.Hash())
+				require.NoError(t, err)
+				if len(receipt.Logs) == 0 {
+					continue
+				}
+				balanceBefore, err := l1Client.BalanceAt(ctx, receipt.Logs[0].Address, new(big.Int).Add(receipt.BlockNumber, big.NewInt(-1)))
+				require.NoError(t, err)
+				balanceAfter, err := l1Client.BalanceAt(ctx, receipt.Logs[0].Address, receipt.BlockNumber)
+				require.NoError(t, err)
+				require.True(t, balanceAfter.Uint64()-balanceBefore.Uint64() == cost.Uint64()*uint64(len(receipt.Logs)), "Cost is mismatch")
+				return true
+			}
+			time.Sleep(time.Second)
```

Comment:
Is the outermost `for loop` and sleep necessary? [This](https://github.com/ethereum-optimism/optimism/blob/a51637c0bb42a2a5f579d0b0ac7a8975f4d47dc5/op-e2e/system/da/eip4844_test.go#L304) example doesn't have this extra level.